src/App.jsx

```diff
@@ -3,46 +3,62 @@ import React, { useState, useEffect } from 'react';
 import AddTodoForm from './AddTodoForm';
 import TodoList from './TodoList';
 
-function useSemiPersistentState() {
-  const [todoList, setTodoList] = useState(
-    JSON.parse(localStorage.getItem('savedTodoList')) || []
-  );
-
-  useEffect(() => {
-    localStorage.setItem('savedTodoList', JSON.stringify(todoList));
-  }, [todoList]);
-
-  return [todoList, setTodoList];
-}
-
 function App() {
-  const [todoList, setTodoList] = useSemiPersistentState();
+  const [todoList, setTodoList] = useState(JSON.parse(localStorage.getItem('todoList')) ?? []);
```

Comment:
I would recommend removing "JSON.parse(localStorage.getItem('todoList')" from your `useState` parameter.  At the moment your application is set up to look for data in an API to populate its list. The purpose of looking into local storage at first was set up as an example and something your application won't be doing. Also, more importantly passing any kind of function directly into your `useState` parameter is not advised since it could trigger some kind of error and you are not gracefully handling any errors. You **could** pass a function into your `useState` parameter that includes a try/catch but that would look messy. So it's best to get the data you need outside of your `useState` and then pass in the data that is returned.

Reply:
You are right, thanks for correcting me, Today I understood that not passing a function directly to the useState parameter could cause errors, since the initial state must be a static value.`  const storedTodoList = JSON.parse(localStorage.getItem('todoList')) || [];
  const [todoList, setTodoList] = useState(storedTodoList);
  const [isLoading, setIsLoading] = useState(true);`